src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProject.cs

```diff
@@ -160,9 +160,12 @@ internal ProjectSystemProject(
         _displayName = displayName;
 
         var provider = _projectSystemProjectFactory.SolutionServices.GetRequiredService<IAnalyzerAssemblyLoaderProvider>();
+
         // NOTE: The provider will always return the same singleton, shadow copying, analyzer loader instance, which is
-        // important to ensure that analyzer dependencies are correctly loaded.
-        _analyzerAssemblyLoader = provider.GetShadowCopyLoader();
+        // important to ensure that analyzer dependencies are correctly loaded.  Note: if we want to support reloading
+        // of analyzer references *within* this workspace (and not just in the OOP roslyn server), this would need to
+        // change to use a dedicate ALC and an IsolatedAssemblyReferenceSet.
+        _analyzerAssemblyLoader = provider.GetSharedShadowCopyLoader();
```

Comment:
My initial gut is this seems surprising to me that we're not making this work, because it _feels_ like if we did, then the special magic around serialization goes away.

Reply:
yeah.  i talked about this with @dibarbet and @ToddGrun .  The intent is to support this in this PR.  i'll do that tomorrow.